typescript/core/src/McapWriter.ts

```diff
@@ -120,7 +129,60 @@ export class McapWriter {
     this.#compressChunk = compressChunk;
   }
 
+  /**
+   * Initializes a new McapWriter in append mode.
+   */
+  static async InitializeInAppendMode({
+    writable,
+    readable,
+  }: {
+    writable: IAppendWritable;
+    readable: IReadable;
+  }): Promise<McapWriter> {
+    const reader = await McapIndexedReader.Initialize({ readable });
+
+    await writable.seek(reader.dataEndOffset!);
+
+    const writer = new McapWriter({
+      writable,
+      appendMode: true,
+    });
+
+    for (const attachmentIndex of reader.attachmentIndexes) {
+      await writer.#addAttachmentIndex(attachmentIndex);
+    }
+
+    for (const metadataIndex of reader.metadataIndexes) {
+      await writer.#addMetadataIndex(metadataIndex);
+    }
+
+    for (const schema of reader.schemasById) {
+      await writer.#registerExistingSchema(schema[1]);
+    }
+
+    for (const channel of reader.channelsById) {
+      await writer.#registerExistingChannel(channel[1]);
+    }
+
+    for (const chunkIndex of reader.chunkIndexes) {
+      await writer.#addChunkIndex(chunkIndex);
+    }
+
+    await writer.#setStatistics(
```

Comment:
I think it would be cleaner and less error-prone if this function could take in the whole Statistics record from the reader as one object instead of individual params.